File: Estrutura_Dados/Python/Pilha.py
class Pilha:
    def __init__(self):
        self.items = []  # Lista usada para armazenar os elementos da pilha

    def isEmpty(self):
        return self.items == []  # Retorna True se a pilha estiver vazia

    def push(self, item):
        self.items.append(item)  # Adiciona um item no topo da pilha

    def pop(self):
        if not self.isEmpty():
            return self.items.pop()  # Remove e retorna o item do topo
        return None  # Retorna None se a pilha estiver vazia

    def peek(self):
        if not self.isEmpty():
            return self.items[-1]  # Retorna o item do topo sem remover
        return None

    def size(self):
        return len(self.items)  # Retorna a quantidade de itens na pilha

File: Estrutura_Dados/Java/Pilha.java
package Estrutura_Dados.Java;

import java.util.LinkedList;

public class Pilha {
    private LinkedList<Object> items;

    // Construtor: inicializa a pilha como uma lista vazia
    public Pilha() {
        items = new LinkedList<>();
    }

    // Verifica se a pilha está vazia
    public boolean isEmpty() {
        return items.isEmpty();
    }

    // Adiciona um item no topo da pilha
    public void push(Object item) {
        items.addLast(item);
    }

    // Remove e retorna o item do topo
    public Object pop() {
        if (!isEmpty()) {
            return items.removeLast();
        }
        return null;
    }

    // Retorna o item do topo sem remover
    public Object peek() {
        if (!isEmpty()) {
            return items.getLast();
        }
        return null;
    }

    // Retorna o tamanho da pilha
    public int size() {
        return items.size();
    }

    // Exibe o conteúdo da pilha
    public void printStack() {
        System.out.println(items);
    }
    
}
